bindings/java/src/main/java/org/apache/opendal/exception/ODException.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.opendal.exception;
+
+public class ODException extends RuntimeException {
+    private final Code code;
+
+    public ODException(byte code, String message) {
```

Comment:
So that we don't have to pass strings from Rust to Java.

Reply:
If we add an error code in rust but forget to add it in Java, this will lead to a wrong error. I prefer to transfer the error code from string. And if there is no suitable error code in Java, a default value (such as `Unkown`) is better than IllegalArgumentException.

Reply:
We can add a test to guard the alignment and fallback to unknown or something on mismatch. Converting from string cannot eliminate the risk for forgetting add the counter part.

Reply:
I mean code of type int is order dependent but the string is not.

Reply:
Yes. That should be preserved by tests. We don't occasionally change enum order, it can be guarded by tests.

See c216fab616da669d93c95649fadc5bd88dba3402